src/Symfony/Bridge/Doctrine/Form/ChoiceList/ORMQueryBuilderLoader.php

```diff
@@ -93,6 +94,14 @@ public function getEntitiesByIds($identifier, array $values)
             $values = array_values(array_filter($values, function ($v) {
                 return (string) $v === (string) (int) $v;
             }));
+        } elseif (in_array($identifierFieldType, array('guid'))) {
```

Comment:
Very good point! I initially (incorrectly) had a check for 'uuid' as well and didn't change to a comparison when I removed that. Will fix.